### PR TITLE
Remove built-in flag from object type

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by zherczeg** [comment]
The main advantage of the patch that it frees 16+2 object types.

**Comment by zherczeg** [comment]
Mostly just noise, small binary size gain, and small stack loss.

Benchmark | Heap (bytes) | Stack (bytes) | Perf (sec) |
----: | ----: | ----: | ----: | 
3d-cube.js | 11184 -> 11184 : 0.000% | 13872 -> 14304 : **-3.114%** | 1.095 -> 1.096 : -0.113% | 
3d-morph.js | 289248 -> 289248 : 0.000% | 1576 -> 1576 : 0.000% | 1.322 -> 1.311 : +0.786% | 
3d-raytrace.js | 110792 -> 110792 : 0.000% | 2564 -> 2612 : **-1.872%** | 1.273 -> 1.272 : +0.049% | 
access-binary-trees.js | 28656 -> 28656 : 0.000% | 3572 -> 3644 : **-2.016%** | 0.752 -> 0.747 : +0.680% | 
access-fannkuch.js | 1808 -> 1808 : 0.000% | 1648 -> 1648 : 0.000% | 2.530 -> 2.550 : -0.807% | 
access-nbody.js | 4600 -> 4600 : 0.000% | 1736 -> 1736 : 0.000% | 1.447 -> 1.444 : +0.149% | 
bitops-3bit-bits-in-byte.js | 1040 -> 1040 : 0.000% | 1592 -> 1592 : 0.000% | 0.703 -> 0.706 : -0.492% | 
bitops-bits-in-byte.js | 1000 -> 1000 : 0.000% | 1592 -> 1592 : 0.000% | 1.003 -> 1.005 : -0.139% | 
bitops-bitwise-and.js | 776 -> 776 : 0.000% | 1344 -> 1344 : 0.000% | 1.264 -> 1.279 : **-1.206%** | 
bitops-nsieve-bits.js | 61200 -> 61200 : 0.000% | 1592 -> 1592 : 0.000% | 1.482 -> 1.499 : **-1.153%** | 
controlflow-recursive.js | 1280 -> 1280 : 0.000% | 61600 -> 63640 : **-3.312%** | 0.517 -> 0.512 : **+1.004%** | 
crypto-aes.js | 28136 -> 28136 : 0.000% | 2312 -> 2344 : **-1.384%** | 0.877 -> 0.885 : -0.937% | 
crypto-md5.js | 70888 -> 70888 : 0.000% | 1720 -> 1760 : **-2.326%** | 0.728 -> 0.728 : -0.115% | 
crypto-sha1.js | 46232 -> 46232 : 0.000% | 1720 -> 1720 : 0.000% | 0.740 -> 0.743 : -0.383% | 
crypto.js | 51624 -> 51624 : 0.000% | 3188 -> 3252 : **-2.008%** | 8.539 -> 8.596 : -0.664% | 
date-format-tofte.js | 8832 -> 8832 : 0.000% | 1992 -> 2024 : **-1.606%** | 0.976 -> 0.976 : -0.038% | 
date-format-xparb.js | 15192 -> 15192 : 0.000% | 2560 -> 2584 : -0.937% | 0.728 -> 0.716 : **+1.550%** | 
deltablue.js | 456304 -> 456304 : 0.000% | 3564 -> 3652 : **-2.469%** | 5.299 -> 5.272 : +0.500% | 
math-cordic.js | 2056 -> 2056 : 0.000% | 1592 -> 1592 : 0.000% | 1.658 -> 1.666 : -0.483% | 
math-partial-sums.js | 1520 -> 1520 : 0.000% | 1568 -> 1568 : 0.000% | 1.141 -> 1.170 : **-2.481%** | 
math-spectral-norm.js | 3240 -> 3240 : 0.000% | 1592 -> 1592 : 0.000% | 0.749 -> 0.753 : -0.592% | 
raytrace.js | 21976 -> 21976 : 0.000% | 5104 -> 5208 : **-2.038%** | 2.560 -> 2.524 : **+1.412%** | 
richards.js | 8464 -> 8464 : 0.000% | 2140 -> 2196 : **-2.617%** | 0.292 -> 0.294 : -0.407% | 
string-base64.js | 90112 -> 90112 : 0.000% | 1640 -> 1640 : 0.000% | 1.433 -> 1.455 : **-1.553%** | 
string-fasta.js | 4408 -> 4408 : 0.000% | 1576 -> 1576 : 0.000% | 3.374 -> 3.381 : -0.214% | 
Geometric mean: | 12101.435 -> 12101.435 : 0.000% | 2495.704 -> 2521.192 : **-1.021%** | 1.251 -> 1.254 : -0.240% | 

Binary (bytes) | master(fea10bb7e3) | patch(34ce2a6dba) | Diff |
----: | ----: | ----: | ----: |
size | 231036 | 231036 | 0 bytes |
.rodata | 20590 | 20590 | 0 bytes |
.dynstr | 384 | 384 | 0 bytes |
.rel.plt | 384 | 384 | 0 bytes |
.interp | 25 | 25 | 0 bytes |
.dynsym | 864 | 864 | 0 bytes |
.gnu.hash | 428 | 428 | 0 bytes |
.text | 204288 | 204232 | -56 bytes |
.comment | 85 | 85 | 0 bytes |
.shstrtab | 226 | 226 | 0 bytes |
.data | 8 | 8 | 0 bytes |
.ARM.exidx | 8 | 8 | 0 bytes |
.rel.dyn | 16 | 16 | 0 bytes |
.init | 12 | 12 | 0 bytes |
.got | 208 | 208 | 0 bytes |
.plt | 608 | 608 | 0 bytes |
.note.ABI-tag | 32 | 32 | 0 bytes |
.gnu.version_r | 64 | 64 | 0 bytes |
.bss | 1575112 | 1575112 | 0 bytes |
.fini | 8 | 8 | 0 bytes |
.hash | 372 | 372 | 0 bytes |
.gnu.version | 108 | 108 | 0 bytes |
.fini_array | 4 | 4 | 0 bytes |
.init_array | 4 | 4 | 0 bytes |
.dynamic | 248 | 248 | 0 bytes |
.eh_frame | 4 | 4 | 0 bytes |
.ARM.attributes | 53 | 53 | 0 bytes |

**Review by rerobika — APPROVED** [review]

File: jerry-core/ecma/operations/ecma-function-object.c [hunk 1]
```diff
@@ -1169,39 +1170,50 @@ ecma_op_function_call_simple (ecma_object_t *func_obj_p, /**< Function object */
...
+ * Perform a native C method call which was registered via the API.
+ *
+ * @return the result of the function call.
+ */
+static ecma_value_t JERRY_ATTR_NOINLINE
+ecma_op_function_call_native (ecma_object_t *func_obj_p, /**< Function object */
+                              ecma_value_t this_arg_value, /**< 'this' argument's value */
+                              const ecma_value_t *arguments_list_p, /**< arguments list */
+                              uint32_t arguments_list_len) /**< length of arguments list */
+
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-function-object.c:1214]: AFAIK, this is a useless empty line

Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-function-object.c]: Thank you. Patch is rebased and fixed.

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**